src/oc_erchef/apps/oc_chef_authz/src/oc_chef_authz_acl.erl

```diff
@@ -329,27 +332,34 @@ convert_actor_ids_to_names(AuthzIds) ->
         oc_chef_group:find_clients_names(AuthzIds, fun chef_sql:select_rows/1),
     {UserNames, DefunctActorAuthzIds} =
         oc_chef_group:find_users_names(RemainingAuthzIds, fun chef_sql:select_rows/1),
-    {ClientNames ++ UserNames, DefunctActorAuthzIds}.
+    {ClientNames, UserNames, DefunctActorAuthzIds}.
+
+ids_to_names(Record, Granular) ->
+    Record1 = process_part(<<"create">>, Record, Granular),
+    Record2 = process_part(<<"read">>, Record1, Granular),
+    Record3 = process_part(<<"update">>, Record2, Granular),
+    Record4 = process_part(<<"delete">>, Record3, Granular),
+    process_part(<<"grant">>, Record4, Granular).
```

Comment:
Almost feels like it should be `process_parts` and it takes in an array of "parts" and grabs the head. I know you didn't write this code so if you don't wanna tackle that refactor no worries, but that would feel more erlang-y.